backend/bytecode.proto

```diff
@@ -0,0 +1,214 @@
+syntax = "proto2";
+
+// x : \tau (used in methods and method types)
+message MethodArgument {
+    required string variable = 1;
+    required Type type = 2;
+}
+
+// Variable declarations are either val or var
+enum VariableDeclarationType {
+    VAL = 1;
+    VAR = 2;
+}
+
+
+// b
+message Bytecode {
+    // v
+    message Version {
+        required int32 magic = 1;
+        required string major = 2;
+        required string minor = 3;
```

Comment:
Hmm, aren't major/minor version numbers usually strings?  They are int16s in Java bytecode